packages/datadog-instrumentations/src/mongodb-core.js

```diff
@@ -0,0 +1,185 @@
+'use strict'
+
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+global.foo = channel('apm:mongodb:insert:start')
+
+addHook({ name: 'mongodb-core', versions: ['2 - 3.1.9'] }, Server => {
+  const serverProto = Server.Server.prototype
+  shimmer.wrap(serverProto, 'command', command => wrapCommand(command, 'command'))
+  shimmer.wrap(serverProto, 'insert', insert => wrapCommand(insert, 'insert', 'insert'))
+  shimmer.wrap(serverProto, 'update', update => wrapCommand(update, 'update', 'update'))
+  shimmer.wrap(serverProto, 'remove', remove => wrapCommand(remove, 'remove', 'remove'))
+
+  const cursorProto = Server.Cursor.prototype
+  shimmer.wrap(cursorProto, '_getmore', _getmore => wrapCursor(_getmore, 'getMore', 'getMore'))
+  shimmer.wrap(cursorProto, '_find', _find => wrapQuery(_find, '_find'))
+  shimmer.wrap(cursorProto, 'kill', kill => wrapCursor(kill, 'killCursors', 'killCursors'))
+  return Server
+})
+
+addHook({ name: 'mongodb', versions: ['>=4'], file: 'lib/cmap/connection.js' }, Connection => {
+  const proto = Connection.Connection.prototype
+  shimmer.wrap(proto, 'command', command => wrapConnectionCommand(command, 'command'))
+  shimmer.wrap(proto, 'query', query => wrapConnectionCommand(query, 'query'))
+  shimmer.wrap(proto, 'getMore', getMore => wrapConnectionCommand(getMore, 'getMore', 'getMore'))
+  shimmer.wrap(proto, 'killCursors', killCursors => wrapConnectionCommand(killCursors, 'killCursors', 'killCursors'))
```

Comment:
Hi @khanayan123 @rochdev 👋 
FYI It seems the instrumentation of `getMore` and `killCursors` is not necessary for mongodb >=4 since the instrumentation of `command` already covers these operations.
I opened an issue a few days ago about this: https://github.com/DataDog/dd-trace-js/issues/1797#issuecomment-1027408097

Reply:
Thanks for the heads up!

Reply:
@khanayan123 Are you addressing this in this PR or are we doing it in another PR?

Reply:
I already removed the wrapping of these functions in the current branch